src/mavoscript.js

```diff
@@ -240,7 +241,18 @@ var _ = Mavo.Script = {
 		},
 		"and": {
 			logical: true,
-			scalar: (a, b) => !!a && !!b,
+			scalar: (a, b) => {
+				if (typeof a !== "boolean" && a !== 0) {
+					return !!b && a;
+				}
+				else if (typeof b !== "boolean" && b !== 0) {
+					return !!a && b;
+				}
+				else {
+					return !!a && !!b;
+				}
```

Comment:
Sorry, I'm not getting how `return val(a)? a : b` would work, since if a is `true` but b is `1` it would return `true` instead of `1`. Or am I missing/misinterpreting something here?

Right now I'm doing something like `return val(a) && val(b) ? retIfTrue : false` and `retIfTrue` is the non-boolean object.

Reply:
Ah yeah, that would be implementing OR, not AND.
For AND, you'd need `!val(a)? a : b`

Reply:
The currently expected output on our tests for `and` is not following how JS `and` works. Should I change it or do I need to add additional logic on `!val(a)? a : b` to get that expected output?

Reply:
Oh is the test always expecting a boolean? Then yes, the test should change.

Reply:
No. If Mavo `and` behaves the same as JS `and`, something like `and([0, 1, 2, 3], [true, true, false])` is returning `[0,true,false,false]` but the expected result right now is `[false, 1, false, false]`.

Reply:
Yeah, I think it should be `[0, 1, false, false]`, and the reversed should be `[0, 1, false, false]`, or whatever JS would produce with a default value of `false` for any missing values.

Reply:
Got it. I made a [new PR](https://github.com/mavoweb/test/pull/11) to fix this.